solutions/src/main/java/ru/sberbank/school/task08/SerializableManager.java

```diff
@@ -17,17 +18,36 @@ public SerializableManager(@NonNull String filesDirectoryPath) {
 
     @Override
     public void initialize() {
-        throw new UnsupportedOperationException("Implement me!");
-    }
+//        throw new UnsupportedOperationException("Implement me!");
+        }
 
     @Override
-    public void saveGame(String filename, Savable gameState) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public void saveGame(@NonNull String filename,
+                         @NonNull MapState<GameObject> gameState) throws SaveGameException {
+        String path = getPath(filename);
+
+        try (FileOutputStream fos = new FileOutputStream(path);
+             ObjectOutputStream out = new ObjectOutputStream(fos)) {
+
+            out.writeObject(gameState);
+        } catch (IOException | NullPointerException e) {
+            throw new SaveGameException("SerializableManager saving error",
+                    e.getCause(), SaveGameException.Type.IO, gameState);
+        }
     }
 
     @Override
-    public Savable loadGame(String filename) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public MapState<GameObject> loadGame(@NonNull String filename) throws SaveGameException {
+        String path = getPath(filename);
+
+        try (FileInputStream fis = new FileInputStream(path);
+             ObjectInputStream in = new ObjectInputStream(fis)) {
```

Comment:
&#39;try resource&#39; has incorrect indentation level 13, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1739996064>*